sdk/tables/aztables/table_pagers.go

```diff
@@ -0,0 +1,442 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package aztables
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"net/http"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/uuid"
+)
+
+// Pager for Table entity queries
+type TableEntityQueryResponsePager interface {
+	// NextPage returns true if the pager advanced to the next page.
+	// Returns false if there are no more pages or an error occurred.
+	NextPage(context.Context) bool
+
+	// Page returns the current TableQueryResponseResponse.
+	PageResponse() TableEntityQueryResponseResponse
+
+	// Err returns the last error encountered while paging.
+	Err() error
+}
+
+type StructEntityQueryResponsePager interface {
```

Comment:
Why do you have the word "Struct" in this interface name?

Reply:
Same question for the other types that include it.

Reply:
To differentiate it from the map version. The naming is really placeholder for now, but I'd be interested to hear any suggestions on how to differentiate these types by name.